### PR TITLE
components: Use updated range styles

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Use the updated range styles from the Global Styles figma: https://www.figma.com/file/oEkcAyhIvPFMVEAO8EImvA/Global-Styles

## How has this been tested?
Run storybook and check out the range control page: `iframe.html?id=components-rangecontrol--default`

### Unfocused

Captura de Tela 2021-08-02 às 10 30 28

### Focused

Captura de Tela 2021-08-02 às 10 30 30

## Types of changes
Style changes (new feature???)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Very nice work, this is working well and is close! This is what I see:

after

Zooming in though, there are a few small things:

Screenshot 2021-08-03 at 11 28 01

The tooltip isn't perfectly centered above the knob. That's probably fine, but if easy to fix would be a nice bit of polish. The focus style is square, though, and 1px. Ideally it's circular like the knob, and 1.5px. This is a bit gnarly as it means we can't use either `outline` or `outline-offset`.

The focus style on the current toggle is actually off (it's 2px instead of 1.5px), but the same CSS could potentially be used:

```
.components-form-toggle .components-form-toggle__input:focus + .components-form-toggle__track {
    box-shadow: 0 0 0 2px #fff, 0 0 0 3.5px var(--wp-admin-theme-color);
    outline: 2px solid transparent;
    outline-offset: 2px;
}
```

I realize that creates a white halo on the inside to emulate the _outset_. I wonder if we can use `background-clip: padding-box;` to accomplish something similar, I recall getting something like that to work for [this scrollbar](https://codepen.io/joen/pen/xxqvjmB) that had a bigger hit-area yet still surfaced the color below. 🤔 

Note that the 2px solid transparent outline is necessary if we use `box-shadow` for the focus style. Windows high contrast mode removes any box shadows entirely, whereas at the same time it makes any `transparent` outlines or borders fully opaque. It's a bit of a trick, but it works well. [More on that here](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/).

Let me know if you'd like me to push some tweaks to the branch directly, happy to take a stab.

File: packages/components/src/range-control/index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function RangeControl(
 		onFocus = noop,
 		onMouseMove = noop,
 		onMouseLeave = noop,
-		railColor,
+		railColor = 'var( --wp-admin-theme-color )',
```
Comment (ciampo) [inline packages/components/src/range-control/index.js]: Not introduced by this PR, but I got a bit confused with the `rail` vs `track` naming. Do they refer to the same part of the UI ?

The README even has an `Anatomy` part where it specifically highlights the "Track", but then there is a prop called `railColor`, a file called `rail.js` and, in the styles, there are CSS snippets refererrin to both the track and the rail.

Should we try to use only one term everywhere?

**Review by ciampo — COMMENTED** [review]
Thank you @sarayourfriend for working on this!

Here's some feedback:

#### Which Figma page?

> Global Styles figma

Which page in the Figma document should be used as the reference?


#### Track color

Compared to what's currently on production, the track seems to show a regression: the track color is always set to the "active" primary color, even for the portion of the track that has values greater than the thumb:

| Production | This PR |
|---|---|
| Screenshot 2021-08-03 at 17 03 40 | Screenshot 2021-08-03 at 17 04 07 |

#### Tooltip centering

Like @jasmussen pointed out, I also noticed that the tooltip isn't aligned with the thumb.

I played around with the code and I noticed that bringing back the `thumbSize` in `packages/components/src/range-control/styles/range-control-styles.js` to its original value of `20` seems to fix the alignment issue — which makes me believe that there are some other hardcoded values in the styles that should also change when the `thumbSize` value changes.

#### Focus styles

This one also has been pointed out by @jasmussen — it'd be great if we could have a circular outline.

Another solution for achieving that may be to add a pseudo-element to the thumb (using `::before` or `::after`) and styling it with a transparent background and a `border`.


#### "With Icon" Storybook examples

Finally, I can't see any icons in the "With Icon Before" and "With Icon After" storybook examples. Are they working as intended?

**Comment by sarayourfriend** [comment]
So interesting, if you check my screenshot of the focus styles... they _are_ circular on Firefox. I'll have to see if Chrome is doing something odd 🤷

File: packages/components/src/range-control/index.js (hunk 1 above)
Comment (sarayourfriend) [reply packages/components/src/range-control/index.js]: Rail is the "filled in" side and "track" is the background.

**Comment by jasmussen** [comment]
> So interesting, if you check my screenshot of the focus styles... they are circular on Firefox. I'll have to see if Chrome is doing something odd 🤷

It appears that the border-radius of the element affects the outline radius on Firefox, but not on Chrome or Safari: 

Screenshot 2021-08-04 at 08 33 38

Not only that, but Firefox even supports subpixel rendering, allowing 1.5px outline widths:

Screenshot 2021-08-04 at 08 33 57

You can test in this codepen: https://codepen.io/joen/pen/LYyJEoL?editors=1100

Since we need both the 1.5px stroke width, and the radius, it seems we need to paint them using box-shadow for the time being until Chrome and Safari catch up. Painting it on a pseudo element seems like it might be a good solution.

**Comment by sarayourfriend** [comment]
Same thing with the track color. From the Screenshots they work fine in Firefox. Seems to be a Chrome thing. I guess I'm glad some people are still using Chrome to catch these shortcomings in that browser 🤷

**Comment by sarayourfriend** [comment]
> Compared to what's currently on production, the track seems to show a regression: the track color is always set to the "active" primary color, even for the portion of the track that has values greater than the thumb:

@ciampo I can't reproduce this on Chrome? Any hints?

**Comment by sarayourfriend** [comment]
@ciampo The icon appears to be an existing bug: https://wordpress.github.io/gutenberg/?path=/story/components-rangecontrol--with-icon-after

**Comment by ciampo** [comment]
> @ciampo I can't reproduce this on Chrome? Any hints?

Not really, I believe I followed the standard steps:

1. Check out the PR
2. `npm run distclean && npm ci && npm run storybook:dev`
3. In Chrome, visit `http://localhost:50240/?path=/story/components-rangecontrol--initial-value-zero`

Screenshot 2021-08-04 at 15 55 35

> The icon appears to be an existing bug

Thank you for the update — we can look into this in a separate PR

File: packages/components/src/range-control/index.js (hunk 1 above)
Comment (ciampo) [reply packages/components/src/range-control/index.js]: Should we add this explanation to the README, maybe adding "Rail" to the "Anatomy" section?

**Comment by sarayourfriend** [comment]
> In Chrome, visit http://localhost:50240/?path=/story/components-rangecontrol--initial-value-zero

Aha, this was the key part, it has something to do with the initial value configuration. I can reproduce it in Firefox now as well, thanks!

**Comment by jasmussen** [comment]
Focus style looks great now, thank you!

focus

I didn't even notice before, but I can also reproduce the fully blue rail color. Just insert a gallery in the post editor and choose the number of columns in the inspector.

**Comment by sarayourfriend** [comment]
Okay the rail color issue is fixed now. I don't really understand why this fixes it though 🤷

**Comment by jasmussen** [comment]
Can confirm, sweet:

Screenshot 2021-08-04 at 16 50 07

Going into super nitpicky details mode, there's a 10px left margin on the item, at least when it's used for the gallery block. Can we remove it?

Screenshot 2021-08-04 at 16 50 41

In extreme detail mode (in fact I had to nearest-neightbor upscale a screenshot to be sure), the range track should be pill shaped, but for whatever reason it _looks_ like it has a 1px border radius:

zoomed

I can see the 3px border radius right there in the inspector, it should for all intents and purposes render the item perfectly pill shaped. But it isn't working for me, and I've no idea why.

---

As a sidenote or little CSS trick I picked up: while a 6px tall rectangle given a 3px border radius should be perfectly pill shaped. However a 6, 7, 10, 12, or 9999px tall rectangle given a _9999px_ will _also_ be perfectly pill-shaped. So long as it's a finite pixel value, border-radius will simply pill-shape the item. While it shouldn't matter in this case, it's a nice simple trick to keep pill-shape CSS simple not needing any math.

**Comment by ciampo** [comment]
> Okay the rail color issue is fixed now. I don't really understand why this fixes it though 🤷

Curious! What was the reason to add a default value in the first place?

**Comment by sarayourfriend** [comment]
> Curious! What was the reason to add a default value in the first place?

I don't remember.

File: packages/components/src/range-control/styles/range-control-styles.js [hunk 2]
```diff
@@ -83,7 +81,7 @@ export const Rail = styled.span`
 	position: absolute;
 	margin-top: 14px;
 	top: 0;
-	border-radius: 3px;
+	border-radius: 9999px;
```
Comment (jasmussen) [inline packages/components/src/range-control/styles/range-control-styles.js:84]: Thanks for doing that!

As it turns out, there appears to be a browser error here, which I will codepen and report to the Chrome team. Spot the odd-one out (open in new tab to see at 1x):

whaaa

**Comment by sarayourfriend** [comment]
@ciampo Do you have a chance to re-review and approve if possible?

**Review by ciampo — APPROVED** [review]
It's looking great, @sarayourfriend !

There are only a few minor things left to improve, feel free to merge this PR after addressing them. 

## Thumb alignment with marks

It looks like the thumb is not centered above the marks. Probably some computation around how the marks are displayed also depends on the `thumbSize`

Screenshot 2021-08-05 at 18 57 53


## Thumb color when `disabled`

This was not necessary with the previous design, since the thumb gray and white. But, now that the thumb is blue, I wonder if we should also change to gray when the input is disabled:

Screenshot 2021-08-05 at 18 56 28


## Bonus round: `Rail` documentation

It'd be great if we added a list item for the `Rail` under the `Anatomy` section in the README.

**Review by ciampo — APPROVED** [review]
The fixes to the thumb disabled color and the thumb-marks alignment are looking good!

**Pull request merged**